xdp.go

```diff
@@ -529,14 +534,34 @@ func (xsk *Socket) Poll(timeout int) (numReceived int, numCompleted int, err err
 }
 
 // GetDescs returns up to n descriptors which are not currently in use.
-func (xsk *Socket) GetDescs(n int) []Desc {
-	if n > len(xsk.freeDescs) {
-		n = len(xsk.freeDescs)
+// if rx is true, return desc in first half of umem, 2nd half otherwise
+func (xsk *Socket) GetDescs(n int, rx bool) []Desc {
+	if n > cap(xsk.getRXDescs) {
+		n = cap(xsk.getRXDescs)
```

Comment:
i notice you're still accessing the RXDesc fields if `rx == false` (here and in the following code) before referencing the TX vars.

I think it might read better to only reference the appropriate fields (depending on `rx`).

Reply:
I assume you mean line 558
		
> start = cap(xsk.getRXDescs)

this is to determine where is start index of TX, the idea  is both freeTXDescs and freeRXDescs has length of options.NumFrames, but freeRXDescs only uses index 0 to options.FillRingNumDescs and freeTXDescs only use options.FillRingNumDesc onward; 
the upside of this approach is not introducing too much changes to the code, the downside is of course part of  freeTXDescs and freeRXDescs are never used; but given these twojust bool slice, so memory waste here is quite small